front/components/assistant/conversation/AgentSuggestion.tsx

```diff
@@ -1,137 +1,141 @@
+      const editedContent = `:mention[${agent.name}]{sId=${agent.sId}} ${userMessage.content}`;
+      const mRes = await fetch(
+        `/api/w/${owner.sId}/assistant/conversations/${conversationId}/messages/${userMessage.sId}/edit`,
+        {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({
+            content: editedContent,
+            mentions: [
+              {
+                type: "agent",
+                configurationId: agent.sId,
+              },
+            ],
+          }),
+        }
+      );
 
-    if (!mRes.ok) {
-      const data = await mRes.json();
-      window.alert(`Error adding mention to message: ${data.error.message}`);
-      sendNotification({
-        type: "error",
-        title: "Invite sent",
-        description: `Error adding mention to message: ${data.error.message}`,
-      });
+      if (!mRes.ok) {
+        const data = await mRes.json();
+        window.alert(`Error adding mention to message: ${data.error.message}`);
+        sendNotification({
+          type: "error",
+          title: "Invite sent",
+          description: `Error adding mention to message: ${data.error.message}`,
+        });
+      }
     }
-  });
+  );
+
+  const agents = useMemo(
+    () =>
+      agentConfigurations.sort((a, b) => {
+        return sortAgents(a, b);
+      }),
+    [agentConfigurations]
+  );
 
   return (
     <div className="pt-4">
-      <div className="text-xs font-bold text-element-600">
-        Which Assistant would you like to talk with?
-      </div>
-      <div className="mt-3 flex flex-col gap-2 sm:flex-row">
-        <Button.List isWrapping={true}>
-          {agents.slice(0, 3).map((agent) => (
-            <div key={`message-${userMessage.sId}-suggestion-${agent.sId}`}>
-              <Button
-                size="xs"
-                variant="avatar"
-                label={`@${agent.name}`}
-                onClick={() => handleSelectSuggestion(agent)}
-                disabled={isSelectingSuggestion}
-                avatar={agent.pictureUrl}
-              />
-            </div>
-          ))}
-          <AssistantPicker
-            owner={owner}
-            assistants={agents.slice(3)}
-            onItemClick={async (agent) => {
-              if (!loading) {
-                setLoading(true);
-                await handleSelectSuggestion(agent);
-                setLoading(false);
-              }
-            }}
-            pickerButton={
-              <Button
-                variant="tertiary"
-                size="xs"
-                icon={RobotIcon}
-                label="Select another"
-              />
+      <div className="flex items-center gap-2">
+        <span className="text-xs font-bold text-element-600">
+          Which Assistant would you like to chat with?
+        </span>
+        <AssistantPicker
+          owner={owner}
+          assistants={agents.slice(3)}
```

Comment:
So we only display 3 assistants in the "AssistantPicker"? It feels like it overlaps with the already displayed assistants, right?

Reply:
We display all assistants but the first 3 in the AssistantPicker

Reply:
Let's split above then in topAgents and rest or something along those lines so it's more explicit.

Reply:
✅